lms/djangoapps/discussion_api/api.py

```diff
@@ -97,14 +117,26 @@ def _cc_thread_to_api_thread(thread):
     }
     ret.update({
         "topic_id": thread["commentable_id"],
+        "group_name": group_ids_to_names.get(thread["group_id"]),
+        "author": None if is_anonymous else thread["username"],
+        "author_label": (
+            None if is_anonymous else
+            "staff" if int(thread["user_id"]) in staff_user_ids else
+            "community_ta" if int(thread["user_id"]) in ta_user_ids else
+            None
+        ),
         "raw_body": thread["body"],
+        "following": thread["id"] in cc_user["subscribed_thread_ids"],
+        "abuse_flagged": cc_user["id"] in thread["abuse_flaggers"],
+        "voted": thread["id"] in cc_user["upvoted_ids"],
+        "vote_count": thread["votes"]["up_count"],
         "comment_count": thread["comments_count"],
         "unread_comment_count": thread["unread_comments_count"],
     })
     return ret
 
 
-def get_thread_list(request, course_key, page, page_size):
+def get_thread_list(request, course, page, page_size):
```

Comment:
Yeah, unfortunately get_cohort_names takes the course object because get_course_cohorts takes the course object because it copies data from the old cohort setting to the new model if necessary.